src/certificate/commands/AdrCertificateCommand.ts

```diff
@@ -0,0 +1,68 @@
+import { Service } from 'typedi';
+import { ITestResult } from '../../models';
+import { ICertificatePayload } from '../../models';
+import { CERTIFICATE_DATA } from '../../models/Enums';
+import { TechRecordType } from '../../models/Types';
+import { TechRecordService } from '../../tech-record/TechRecordService';
+import { ICertificatePayloadCommand } from '../ICertificatePayloadCommand';
+
+@Service()
+export class AdrCertificateCommand implements ICertificatePayloadCommand {
+	private type?: CERTIFICATE_DATA;
+
+	constructor(private techRecordService: TechRecordService) {}
+
+	private certificateIsAnAdr = (): boolean => this.type === CERTIFICATE_DATA.ADR_DATA;
+
+	public initialise(type: CERTIFICATE_DATA, isWelsh = false) {
```

Comment:
is welsh isn't being used in a lot of these commands, can they be removed?

Reply:
So `isWelsh` is used by some of the commands and so it exists on the initialise for them all to maintain the same interface. We want this because....take a look at how we iterate the collection in https://github.com/dvsa/cvs-tsk-cert-gen/blob/56df664faa5cfc540e45485df7762fc7f7e9a2a7/src/certificate/CertificatePayloadGenerator.ts#L74.

If we remove `isWelsh` for some of the commands we would need to individually initialise / generate each certificate command which isn't particularly DRY, especially when I feel it's a reasonable assertion that `isWelsh` may be used across those commands in the future. yk?

Reply:
Thinking about this a bit more, it's fairly typical here to use a StageBag for injecting params...we could refactor that way. It wouldn't have any real functional difference, but would probably read more cleanly...which is really the intended outcome here.

Reply:
So I have introduced the state bag pattern and refactored the commands to use a base abstract class to make each command more DRY.